src/app/modules/sgid/sgid.service.ts

```diff
@@ -40,10 +40,13 @@ export class SgidServiceClass {
     cookieMaxAgePreserved,
     privateKeyPath,
     publicKeyPath,
+    hostname,
     ...sgidOptions
   }: ISgidVarsSchema) {
     this.privateKey = fs.readFileSync(privateKeyPath, { encoding: 'utf8' })
     this.client = new SgidClient({
+      // If hostname is empty, use the default provided by sgid-client.
+      hostname: hostname || undefined,
```

Comment:
/justMusingNoNeedToDoAntyhing: `|| undefined` feels a bit "funny". It's there to account for when`sgid.config` itself defaulted to empty string, correct? Could we not just removed the default as empty string in config, or default it to `undefined` there directly?

Reply:
i wanted to default it to undefined in the config, but convict wouldn't let me do it :( actually let me play around with this to see if it'll let me do it. I completely agree that this is weird haha

Reply:
<img width="1494" alt="image" src="https://user-images.githubusercontent.com/25571626/210027745-6ccd8aea-73ed-4f12-98a4-95a862d14827.png">

seems like convict strictly doesn't allow `default` to be undefined, so we have to specify something. Most people seem to be using `nullable` property, but that just sets the default value to `null`, not `undefined`, so anyway we would have to do this check haha